Content.Shared/Climbing/Systems/ClimbSystem.cs

```diff
@@ -157,327 +157,335 @@ private void OnParentChange(EntityUid uid, ClimbingComponent component, ref EntP
+        if (args.Handled || args.User != args.Dragged && !HasComp<HandsComponent>(args.User))
+            return;
+
+        TryClimb(args.User, args.Dragged, uid, out _, component);
+    }
+
+    public bool TryClimb(
+        EntityUid user,
+        EntityUid entityToMove,
+        EntityUid climbable,
+        out DoAfterId? id,
+        ClimbableComponent? comp = null,
+        ClimbingComponent? climbing = null)
+    {
+        id = null;
+
+        if (!Resolve(climbable, ref comp) || !Resolve(entityToMove, ref climbing))
+            return false;
+
+        // Note, IsClimbing does not mean a DoAfter is active, it means the target has already finished a DoAfter and
+        // is currently on top of something..
+        if (climbing.IsClimbing)
+            return true;
+
+        var args = new DoAfterArgs(EntityManager, user, comp.ClimbDelay, new ClimbDoAfterEvent(),
+            entityToMove,
+            target: climbable,
+            used: entityToMove)
+        {
+            BreakOnTargetMove = true,
+            BreakOnUserMove = true,
+            BreakOnDamage = true
+        };
+
+        _audio.PlayPredicted(comp.StartClimbSound, climbable, user);
+        return _doAfterSystem.TryStartDoAfter(args, out id);
+    }
+
+    private void OnDoAfter(EntityUid uid, ClimbingComponent component, ClimbDoAfterEvent args)
+    {
+        if (args.Handled || args.Cancelled || args.Args.Target == null || args.Args.Used == null)
+            return;
+
+        Climb(uid, args.Args.User, args.Args.Target.Value, climbing: component);
+        args.Handled = true;
+    }
+
+    private void Climb(EntityUid uid, EntityUid user, EntityUid climbable, bool silent = false, ClimbingComponent? climbing = null,
+        PhysicsComponent? physics = null, FixturesComponent? fixtures = null, ClimbableComponent? comp = null)
+    {
+        if (!Resolve(uid, ref climbing, ref physics, ref fixtures, false))
+            return;
+
+        if (!Resolve(climbable, ref comp))
+            return;
+
+        if (!ReplaceFixtures(uid, climbing, fixtures))
+            return;
+
+        var xform = _xformQuery.GetComponent(uid);
+        var (worldPos, worldRot) = _xformSystem.GetWorldPositionRotation(xform);
+        var worldDirection = _xformSystem.GetWorldPosition(climbable) - worldPos;
+        var distance = worldDirection.Length();
+
+        climbing.IsClimbing = true;
+        var climbDuration = TimeSpan.FromSeconds(distance / climbing.TransitionRate);
+        climbing.NextTransition = _timing.CurTime + climbDuration;
+
+        if (distance > 0)
+        {
+            var parentRot = (worldRot - xform.LocalRotation);
+            // Need direction relative to climber's parent.
+            var localDirection = (-parentRot).RotateVec(worldDirection);
+            climbing.Direction = localDirection.Normalized() * climbing.TransitionRate;
+        }
+        else
+        {
+            climbing.Direction = Vector2.Zero;
+        }
```

Comment:
Does this just stun them if it's a 0-length vector.

Reply:
No, because climb duration depends on a distance too

https://github.com/space-wizards/space-station-14/assets/16546901/6b5eb89e-2c50-4df2-8896-5ba01a32771f

Reply:
Actually nvm maybe it does for one tick